### DISCUSSION TITLE
ESP32 With Ethernet, Best Board So Far With MicroPython?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#10487] [General]
ESP32 With Ethernet, Best Board So Far With MicroPython?  I have used ENC28j60 Ethernet breakout boards with ESP8266 and ESP32 modules, but that was more fun and games than serious use.  Now, I have a project that is more important, where I might consider using  Pi, I am looking for an alternate that has ethernet, and can host as simple web server, a bit like Tasmota but not as complex, other than driving a few GPIOs with PWM, and maybe an additional GPIO for a status LED (separate from on-board), not much more.

**Comment by jimmo** [comment]
What's your actual question? Are you looking for an ESP32 board with Ethernet support (there are lots of options including the Ethernet DevKit from Espressif), but it depends if you want ESP32 or S2/S3, etc. Or are you looking for a RP2040 board? The WizNet ones in the Pico form factor seem to work well.

**Comment by robert-hh** [comment]
Olimex sells ESP32 boards with Ethernet and POE support. The boards use a WROOM module (4MB flash, standard RAM). And there are for instance the Wiznet pico boards with Ethernet, You can as well use a TEENSY4.1 with RJ45 plug, or other mimxrt boards or STM32 boards with ethernet.

**Comment by Jibun-no-Kage** [comment]
At this point just asking what others have used, liked.  As I noted I have used en28j60 boards with various ESP modules in the past, but this time thinking some integrated is preferred.
